test-suite/src/test/scala/org/scalajs/testsuite/javalib/ListTest.scala

```diff
@@ -0,0 +1,348 @@
+/*                     __                                               *\
+**     ________ ___   / /  ___      __ ____  Scala.js Test Suite        **
+**    / __/ __// _ | / /  / _ | __ / // __/  (c) 2013, LAMP/EPFL        **
+**  __\ \/ /__/ __ |/ /__/ __ |/_// /_\ \    http://scala-js.org/       **
+** /____/\___/_/ |_/____/_/ | |__/ /____/                               **
+**                          |/____/                                     **
+\*                                                                      */
+package org.scalajs.testsuite.javalib
+
+import scala.language.implicitConversions
+
+import org.scalajs.jasminetest.JasmineTest
+
+trait ListTest[F <: ListFactory] extends JasmineTest {
+
+  def listFactory: F
+
+  def expectIndexOutOfBoundsException[T](f: => T): Unit = {
+    expect({
+      try {
+        f
+        false
+      } catch {
+        case err: IndexOutOfBoundsException => true
+      }
+    }).toBeTruthy
+  }
+
+  def testListApi(): Unit = {
+    it("should store strings") {
+      val al = listFactory.empty[String]
+
+      expect(al.size()).toEqual(0)
+      al.add("one")
+      expect(al.size()).toEqual(1)
+      expect(al.get(0)).toEqual("one")
+      al.add("two")
+      expect(al.size()).toEqual(2)
+      expect(al.get(0)).toEqual("one")
+      expect(al.get(1)).toEqual("two")
+
+      expectIndexOutOfBoundsException(al.get(-1))
+      expectIndexOutOfBoundsException(al.get(al.size))
+    }
+
+    it("should store integers") {
+      val al = listFactory.empty[Int]
+
+      al.add(1)
+      expect(al.size()).toEqual(1)
+      expect(al.get(0)).toEqual(1)
+      al.add(2)
+      expect(al.size()).toEqual(2)
+      expect(al.get(0)).toEqual(1)
+      expect(al.get(1)).toEqual(2)
+
+      expectIndexOutOfBoundsException(al.get(-1))
+      expectIndexOutOfBoundsException(al.get(al.size))
+    }
+
+    it("should store doubles") {
+      val al = listFactory.empty[Double]
+
+      al.add(1.234)
+      expect(al.size()).toEqual(1)
+      expect(al.get(0)).toEqual(1.234)
+      al.add(2.345)
+      expect(al.size()).toEqual(2)
+      expect(al.get(0)).toEqual(1.234)
+      expect(al.get(1)).toEqual(2.345)
+      al.add(Double.NaN)
+      al.add(+0.0)
+      al.add(-0.0)
+      expect(al.size()).toEqual(5)
+      expect(al.get(0)).toEqual(1.234)
+      expect(al.get(1)).toEqual(2.345)
+      expect(al.get(2).isNaN).toBeTruthy
+      expect(al.get(3)).toEqual(+0.0)
+      expect(al.get(4)).toEqual(-0.0)
```

Comment:
To test the difference between `+0.0` and `-0.0`, use this:
```scala
expect(al.get(3).equals(+0.0)).toBeTruthy
```
`toEqual` would succeed even when comparing `+0.0` to `-0.0`.